miso-web/src/main/webapp/scripts/hot_sample.js

```diff
@@ -0,0 +1,490 @@
+        
+        // Stock columns
+        HotUtils.makeColumnForEnum('STR Status', show['Stock'], true, 'strStatus', Constants.strStatuses),
+        {
+          header: 'DNAse',
+          data: 'dnaseTreated',
+          type: 'dropdown',
+          trimDropdown: false,
+          validator: HotUtils.validator.requiredAutocomplete,
+          source: [ 'True', 'False' ],
+          unpack: function (sam, flat, setCellMeta) {
+            flat.dnaseTreated = (sam.dnaseTreated ? 'True' : 'False');
+          },
+          pack: function (sam, flat, errorHandler) {
+            sam.dnaseTreated = flat.dnaseTreated === 'True';
+          },
+          include: show['Stock'] && config.targetSampleClass.dnaseTreatable
+        },
+        HotUtils.makeColumnForFloat('Vol. (&#181;l)', (show['Stock'] || show['Aliquot']), 'volume'),
+        HotUtils.makeColumnForFloat('Conc. (ng/&#181;l)', (show['Stock'] || show['Aliquot']), 'concentration'),
+        HotUtils.makeColumnForFloat('New RIN', config.rnaSamples, 'rin'),
+        HotUtils.makeColumnForFloat('New DV200', config.rnaSamples, 'dv200'),
+        
+        // QC status columns for detailed and non-detailed samples
+        {
+          header: 'QC Passed?',
+          data: 'qcPassed',
+          type: 'dropdown',
+          trimDropdown: false,
+          source: ['Unknown', 'True', 'False'],
+          unpack: function (sam, flat, setCellMeta) {
+            if (sam.qcPassed === true) flat.qcPassed = 'True';
+            else if (sam.qcPassed === false) flat.qcPassed = 'False';
+            else flat.qcPassed = 'Unknown';
+          },
+          pack: function (sam, flat, errorHandler) {
+            if (flat.qcPassed === 'True') sam.qcPassed = true;
+            else if (flat.qcPassed === 'False') sam.qcPassed = false;
+            else sam.qcPassed = null;
+          },
+          include: !Constants.isDetailedSample
+        },
+        {
+          header: 'QC Status',
+          data: 'detailedQcStatusDescription',
+          type: 'dropdown',
+          trimDropdown: false,
+          source: (function () {
+            var statuses = Constants.detailedQcStatuses.sort(function (a, b) {
+              function statusToInt (status) {
+                if (status === true) return 0;
+                if (status === false) return 1;
+                return 2;
+              }
+              return statusToInt(a.status) - statusToInt(b.status);
+            }).map(function (s) { return s.description; });
+            statuses.unshift('Not Ready'); // can't chain this because unshift returns the length of the array
+            return statuses;
+          })(),
+          validator: HotUtils.validator.requiredText,
+          unpack: function (sam, flat, setCellMeta) {
+            flat.detailedQcStatusDescription = Utils.array.maybeGetProperty(
+                Utils.array.findFirstOrNull(
+                    Utils.array.idPredicate(sam.detailedQcStatusId), Constants.detailedQcStatuses),
```

Comment:
That's fine on create, but on update too? That means you have to choose from the dropdown every time you bulk edit samples that were set to 'Not Ready.' That's confusing if you were only intending to update other fields